### PR TITLE
[WEB-1133] clinics updates

### PR DESCRIPTION
**Pull request by krystophv**
For [WEB-1133], updates clinic API endpoints
Associated with https://github.com/tidepool-org/blip/pull/900

[WEB-1133]: https://tidepool.atlassian.net/browse/WEB-1133

### CODE REVIEW

File: clinics.js [hunk 1]
```diff
@@ -168,35 +192,46 @@ module.exports = function (common) {
...
      *
-     * @param {Number} clinicId - Id of the clinic
-     * @param {Object} patient
-     * @param {String} [patient.id] - Id of this relationship
-     * @param {String} [patient.patientId] - Id of the patient
-     * @param {String} [patient.clinicId] - Id of clinic
-     * @param {String[]} [patient.permissions] - Array of string permissions
+     * @param {String} clinicId - ID of the clinic
+     * @param {Object} patient - new patient
+     * @param {String} patient.email - The email address of the patient
```
Comment (cbwebdevelopment) [inline clinics.js:224]:
```suggestion
     * @param {String} [patient.email] - The email address of the patient
```

File: clinics.js [hunk 2]
```diff
@@ -168,35 +192,46 @@ module.exports = function (common) {
...
+     * @param {String} patient.fullName - The full name of the patient
```
Comment (cbwebdevelopment) [inline clinics.js:225]:
```suggestion
     * @param {String} [patient.fullName] - The full name of the patient
```

File: clinics.js [hunk 3]
```diff
@@ -168,35 +192,46 @@ module.exports = function (common) {
...
+     * @param {String} patient.birthDate - YYYY-MM-DD
```
Comment (cbwebdevelopment) [inline clinics.js:226]:
```suggestion
     * @param {String} [patient.birthDate] - YYYY-MM-DD
```

File: clinics.js [hunk 4]
```diff
@@ -223,136 +258,228 @@ module.exports = function (common) {
...
      * Modify a patient from clinic
      *
-     * @param {Number} clinicId - Id of the clinic
-     * @param {Number} patientId - Id of the patient
-     * @param {Object} updates
-     * @param {String[]} updates.permissions - Array of string permissions
+     * @param {String} clinicId - Id of the clinic
+     * @param {String} patientId - Id of the patient
+     * @param {Object} patient - new patient
+     * @param {String} patient.email - The email address of the patient
```
Comment (cbwebdevelopment) [inline clinics.js:264]:
```suggestion
     * @param {String} [patient.email] - The email address of the patient
```

File: clinics.js [hunk 5]
```diff
@@ -223,136 +258,228 @@ module.exports = function (common) {
...
+     * @param {String} patient.fullName - The full name of the patient
```
Comment (cbwebdevelopment) [inline clinics.js:265]:
```suggestion
     * @param {String} [patient.fullName] - The full name of the patient
```

File: clinics.js [hunk 6]
```diff
@@ -223,136 +258,228 @@ module.exports = function (common) {
...
+     * @param {String} patient.birthDate - YYYY-MM-DD
```
Comment (cbwebdevelopment) [inline clinics.js:266]:
```suggestion
     * @param {String} [patient.birthDate] - YYYY-MM-DD
```

File: clinics.js [hunk 7]
```diff
@@ -223,136 +258,228 @@ module.exports = function (common) {
...
 
     /**
-     * Remove association of clinic to patient
+     * Send an invite to a clinician to join a clinic
      *
-     * @param {Number} clinicId - Id of the clinic
-     * @param {Number} patientId - Id of the patient
+     * @param {String} clinicId - clinic ID
+     * @param {Object} clinician - clinician Invite object
+     * @param {String} clinician.email - clinician's email address
```
Comment (cbwebdevelopment) [inline clinics.js:287]:
```suggestion
     * @param {String} [clinician.email] - clinician's email address
```

File: clinics.js [hunk 8]
```diff
@@ -223,136 +258,228 @@ module.exports = function (common) {
...
+     * @param {String[]} clinician.roles - array of clinician's roles
```
Comment (cbwebdevelopment) [inline clinics.js:288]:
```suggestion
     * @param {String[]} [clinician.roles] - array of clinician's roles
```

File: clinics.js [hunk 9]
```diff
@@ -223,136 +258,228 @@ module.exports = function (common) {
...
      */
-    addClinicianToClinic: function(clinicId, clinician, cb){
+    acceptPatientInvitation: function(clinicId, inviteId, cb){
       common.assertArgumentsSize(3);
-      common.doPostWithToken(
-        `/clinics/${clinicId}/clinicians`,
-        clinician,
+      common.doPutWithToken(
+        `/v1/clinics/${clinicId}/invites/patients/${inviteId}`,
+        '',
```
Comment (cbwebdevelopment) [inline clinics.js:362]: I'm curious why we're passing an empty arg here.  I'm assuming the backend has a default, but allows this parameter to contain other values.

**Review by cbwebdevelopment — APPROVED** [review]
LGTM 🥇 
Great job on this - especially on documenting it all.  A lot more new api work than I'd imagined.

I have a few suggested edits to consolidate the formatting of the nested params, and a general curiosity question about an empty arg we're sending in a couple of calls, but that's it.

File: clinics.js [hunk 10]
```diff
@@ -35,19 +33,23 @@ module.exports = function (common) {
...
+     * @param {String} [clinic.country] - Clinic country
      * @param {Object[]} [clinic.phoneNumbers] - Array of phone number objects for the clinic
      * @param {String} [clinic.phoneNumbers[].type] - Phone number description
      * @param {String} [clinic.phoneNumbers[].number] - Phone number
-     * @param {String} [clinic.location] - Location
-     * @param {Object} [clinic.metadata]
-     *
+     * @param {String} [clinic.clinicType] - Type of clinic
+     * @param {Number} [clinic.clinicSize] - Patient population size
+     * @param {String} clinic.email - Primary email address for clinic
```
[the remark this answers is not in the document]
Comment (toddkazakov) [reply clinics.js:45]: @krystophv just fyi, that we'll be dropping the clinic email in favour of the share code - this is what we discussed few weeks ago.

File: clinics.js (hunk 9 above)
Comment (krystophv) [reply clinics.js:362]: The API spec details that this endpoint doesn't expect anything for the `body` of the `PUT`, but the format of the `doPutWithToken` essentially requires a `data` that will be sent to the server in order to not stomp on the `codes`/`cb` params. Do you think that going with `null` as opposed to the empty string would be better/more explicit (like `confirm.removeInvite`)?

Comment (cbwebdevelopment) [reply clinics.js:362]: Using `null` does seem more fitting to me.

**Pull request merged**